universe/Building.h

```diff
@@ -214,9 +214,8 @@ class BuildingTypeManager {
 
 private:
     BuildingTypeManager();
-    ~BuildingTypeManager();
 
-    std::map<std::string, BuildingType*> m_building_types;
+    std::map<std::string, std::unique_ptr<BuildingType>> m_building_types;
```

Comment:
To be clear, I'm / we're talking about two distinct instances of `shared_ptr` that point to / share ownership of the same object, being dereferenced to call a const member function of the object, or to read a POD / built-in type like int, with no modification of the `shared_ptr` or the pointed-to object occurring.

According to http://en.cppreference.com/w/cpp/memory/shared_ptr all the member functions can be called on separate instances of a `shared_ptr` even if they point to the same object. Presumably this includes `get()`, so whatever syncronization issues you're referring to must also apply to raw pointers as well then... (?)

The scenario I'm asking about is two threads accessing the same object through two distinct pointers that point to that same object. The access is only using const member functions or simple reads of the value of the pointed-to data, which don't do anything internal with mutable state or otherwise alter the state of the object or data.

You're saying that's not thread safe and can crash?